src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,139 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_LIST_SIZE = 10;
+    private static final int DEFAULT_MULTIPLIER = 1;
+    private int size;
+    private T[] internalArray;
+
+    public ArrayList() {
+        internalArray = (T[]) new Object[DEFAULT_LIST_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= 0 && size < internalArray.length) {
```

Comment:
use checkIndex here

Reply:
It seems to me that checkIndex() method cannot be applied here.

Reply:
Are you expecting size < 0 in some cases here?
Check for **grow** and add new value. Method body should have 4 lines. Or even 2))